### PR TITLE
Improvements on `CollectionView`, resolution of bug and API change.

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

The work done recently for `ReentrantCollectionView` has revealed some similar issues with
`CollectionView`, which we now correct.

## Proposal

The following was done:
* The `load_entry` has been renamed as `load_entry_or_insert`.
* The `contains_key` is added to the `CollectionView`.
* The `try_load_entry` is now a true `&self` function (e.g. the hash does not change) though the interior mutability is still needed.
* The `try_load_entry` is now returning an `Option<ReadGuardedView<W>>`.

The function `try_load_entry` has been added to the tests.

## Test Plan

The CI test has been improved.

## Release Plan

No change. The documentation of the library is changed accordingly.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-views/src/collection_view.rs [hunk 1]
```diff
@@ -225,34 +234,28 @@ where
...
+                let key_index = self.context.base_tag_index(KeyTag::Index as u8, &short_key);
+                if !self.delete_storage_first && self.context.contains_key(&key_index).await? {
+                    let key = self
+                        .context
+                        .base_tag_index(KeyTag::Subview as u8, &short_key);
+                    let context = self.context.clone_with_base_key(key);
+                    let mut view = W::load(context).await?;
+                    if self.delete_storage_first {
+                        view.clear();
+                    }
```
Comment (ma2bd) [inline linera-views/src/collection_view.rs:252]: These are unreachable

**Review by ma2bd — APPROVED** [review]
Good catch! Thanks for the follow up

File: linera-views/src/collection_view.rs (hunk 1 above)
Comment (MathieuDutSik) [reply linera-views/src/collection_view.rs:252]: Thank you for the catch.

File: linera-views/src/collection_view.rs [hunk 2]
```diff
@@ -281,6 +281,36 @@ where
...
+    ///   assert!(view.contains_key(&[0, 1]).await.unwrap());
+    ///   assert!(!view.contains_key(&[0, 2]).await.unwrap());
+    /// # })
+    /// ```
+    pub async fn contains_key(&self, short_key: &[u8]) -> Result<bool, ViewError> {
+        let updates = self.updates.write().await;
+        Ok(match updates.get(short_key) {
+            Some(entry) => match entry {
+                Update::Set(_view) => true,
+                _entry @ Update::Removed => false,
```
Comment (afck) [inline linera-views/src/collection_view.rs:305]: (Unnecessary `_entry @`.)

**Pull request merged**